designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessingTypeDeploymentService.scala

```diff
@@ -29,8 +29,11 @@ final case class DeployedScenarioData(
     resolvedScenario: CanonicalProcess
 )
 
-class ProcessingTypeDeploymentServiceStub(deployedScenarios: List[DeployedScenarioData])
-    extends ProcessingTypeDeploymentService {
+class ProcessingTypeDeploymentServiceStub(
+    deployedScenarios: List[DeployedScenarioData],
+) extends ProcessingTypeDeploymentService {
+
+  protected val actionIds: scala.collection.mutable.Buffer[ProcessActionId] = scala.collection.mutable.Buffer.empty
```

Comment:
It isn't thread safety and I'm not sure if we aren't doing something concurrently here. I'd recommend to replace this by `@volatile var` with immutable List